virttest/utils_net.py

```diff
@@ -3080,6 +3080,30 @@ def generate_mac_address_simple():
     return mac
 
 
+def gen_ipv4_addr(network_num="10.0.0.0", network_prefix="24", exclude_ips=[]):
+    """
+    generate ipv4 address
+    :param network_num: network number to be used
+    :param network_prefix: prefix used to get subnet mask to calculate ip range
+    :param exclude_ips: the list of ipaddress should be excluded
+
+    :return: ipaddress of type str
+    """
+    ip_regex = "^\d+.\d+.\d+.\d+$"
+    if not re.match(ip_regex, network_num):
+        network_num = "10.0.0.0"
+    if not exclude_ips:
+        exclude_ips.append(network_num)
+        exclude_ips.append('.'.join(network_num.split('.')[0:3]) + ".%s" %
+                           str(1))
+        exclude_ips.append(('.'.join(network_num.split('.')[0:3]) + ".%s" %
+                           str(255)))
```

Comment:
this is for default value when `network_prefix = "24"`, if user gives different network prefix for his required subnet, then user can also give `exclude_ips` too.